my-app/src/components/AutoLogout.tsx

```
@@ -0,0 +1,136 @@
+
+    //logout after 5 mins
+    timeoutRef.current = setTimeout(() => {
+      signOut(auth);
+      setType(AutoLogoutType.LOGOUT);
+    }, FIVE_MINUTES);
+  };
+
+  useEffect(() => {
+    const activityEvents = ["keydown", "click", "scroll", "touchstart"];
+
+    const handleActivity = useCallback(() => {
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)